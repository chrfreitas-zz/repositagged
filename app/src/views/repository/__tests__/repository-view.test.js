import React from 'react';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';
import Repository from '../index';

describe('Repository View', () => {
  const props = {
    repository: {
      id: 1,
    },
    update: () => {},
  };

  it('should match with snapshot', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Repository {...props} />);
    expect(result).toMatchSnapshot();
  });

  it('should update the repository object', () => {
    props.update = jest.fn();
    const wrapper = shallow(<Repository {...props} />);

    wrapper.instance().edit('eml');
    expect(props.update).toHaveBeenCalledWith({ id: 1, tags: 'eml' });
  });
});
