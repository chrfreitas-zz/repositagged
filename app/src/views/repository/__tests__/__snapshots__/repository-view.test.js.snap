// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository View should match with snapshot 1`] = `
<ModalTags
  close={[Function]}
  isOpen={true}
  repository={
    Object {
      "id": 1,
    }
  }
  save={[Function]}
/>
`;
